api/v1alpha6/openstackcluster_conversion.go

```diff
@@ -164,23 +164,14 @@ func restorev1alpha6ClusterSpec(previous *OpenStackClusterSpec, dst *OpenStackCl
 func restorev1beta1ClusterSpec(previous *infrav1.OpenStackClusterSpec, dst *infrav1.OpenStackClusterSpec) {
 	// Bastion is restored separately
 
-	if dst.Network.IsEmpty() {
-		dst.Network = previous.Network
-	}
+	restorev1beta1NetworkParam(previous.Network, dst.Network)
 
-	// Restore all fields except ID, which should have been copied over in conversion
+	// ExternalNetwork by filter will be been lost in down-conversion
```

Comment:
Tiny nit, not worth a new push IMO, just mentioning since I happened to see it
```suggestion
	// ExternalNetwork by filter will be lost in down-conversion
```